src/Proto.Cluster/Messages/Messages.cs

```diff
@@ -58,4 +58,46 @@ public static uint GetMembershipHashCode(IEnumerable<Member> members)
             return hash;
         }
     }
+
+    public partial class GossipKeyValue
+    {
+        public string GlobalKey => $"{MemberId}.{Key}";
+    }
+
+    public partial class GossipState
+    {
+        public (bool dirty, GossipState state) MergeWith(GossipState other)
```

Comment:
Merge two GossipState objects. return the new state and bool dirty if anything changed